src/main/java/org/htmlunit/html/HtmlFormUtil.java

```diff
@@ -98,13 +98,18 @@ public static List<HtmlElement> getSubmitButtons(final HtmlForm htmlForm) throws
     }
 
     /**
-     * Gets the first {@code <input type="submit">} element in this form.
+     * Gets the first {@code <input type="submit">} or {@code <button name="Submit">} element in this form.
      */
     public static HtmlElement getSubmitButton(final HtmlForm htmlForm) throws ElementNotFoundException {
         List<HtmlElement> submitButtons = getSubmitButtons(htmlForm);
         if (!submitButtons.isEmpty()) {
             return submitButtons.get(0);
         }
+        for (HtmlElement element : htmlForm.getElementsByAttribute("button", "name", "Submit")) {
+            if(element instanceof HtmlButton) {
+                return element;
+            }
+        }
```

Comment:
Without this, any other button that happens to show first in the page is returned, which is likely not going to be the Submit (Save) button for the form.

Reply:
I think this works but may still be brittle, as the submit button could be named something else.
(e.g.  Deploy plugin in plugin manager).  -> https://github.com/jenkinsci/jenkins/blob/master/test/src/test/java/hudson/PluginManagerTest.java#L152-L159 

So I think this may be better looking to find this submit first, then a button that is [primary](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/resources/lib/form/submit.jelly#L44-L49), and then fall back to the first button in the form?

Reply:
Looks fragile, as the submit button name can be customized.
 
As a workaround, ignore button with class `jenkins-copy-button` and we would need to add a new class e.g. `jenkins-submit-button` to https://github.com/jenkinsci/jenkins/blob/58a8f2e018f733cf4ff9155f2c4806e079cbd31d/core/src/main/resources/lib/form/submit.jelly#L57

Reply:
> As a workaround, ignore button with class jenkins-copy-button

This would solve the specific case of `JenkinsRule#configRoundtrip` (the "copy" button in the one under `/configure` which copies the `JENKINS_HOME` path to the clipboard) but not other potential cases where there other buttons. So, as a workaround (until the core patch), I think @jtnord proposal might be the best (or less bad) one?

Reply:
Also, the _default_ value is actually "Submit", so I think this would be dodging most of the possible hits?

Reply:
you are correct, I was confusing the text and the name.